sw/device/lib/crypto/impl/ecc.c

```diff
@@ -213,3 +412,310 @@ crypto_status_t otcrypto_x25519_async_finalize(
   // TODO: X25519 is not yet implemented.
   return kCryptoStatusNotImplemented;
 }
+
+/**
+ * Consistency checks for caller-provided private key configurations.
+ *
+ * This check ensures:
+ *   - The key length in the configuration matches the given curve
+ *   - The key mode matches expectations.
+ *
+ * @param elliptic_curve Curve for which key is intended.
+ * @param config Private key configuration.
+ * @param expected_mode Expected key mode.
+ * @returns OK if the check passes, BAD_ARGS otherwise.
+ */
+static status_t key_config_check(const ecc_curve_t *elliptic_curve,
+                                 crypto_key_config_t *config,
+                                 key_mode_t expected_mode) {
+  // Check the key mode.
+  if (private_key->config.key_mode != expected_mode) {
```

Comment:
Is this just `config->key_mode` now? Same comment for all occurrences of `private_key` in this function.